qiskit_ibm_runtime/noise_learner.py

```diff
@@ -0,0 +1,276 @@
+                If neither is specified, and the primitive is created inside a
+                :class:`qiskit_ibm_runtime.Session` context manager, then the session is used.
+                Otherwise if IBM Cloud channel is used, a default backend is selected.
+
+            options: :class:`NoiseLearnerOptions`. Alternatively, :class:`EstimatorOptions` can be
+                provided, in which case the estimator options get reformatted into noise learner options
+                and all the irrelevant fields are ignored.
+        """
+        self._mode: Optional[Union[Session, Batch]] = None
+        self._service: QiskitRuntimeService | QiskitRuntimeLocalService = None
+        self._backend: Optional[BackendV1 | BackendV2] = None
+
+        if isinstance(mode, (Session, Batch)):
+            self._mode = mode
+            self._service = self._mode.service
+            self._backend = self._mode._backend
+        elif isinstance(mode, IBMBackend):  # type: ignore[unreachable]
+            self._service = mode.service
+            self._backend = mode
+        elif isinstance(mode, (BackendV1, BackendV2)):
+            self._service = QiskitRuntimeLocalService()
+            self._backend = mode
+        elif isinstance(mode, str):
+            self._service = (
+                QiskitRuntimeService()
+                if QiskitRuntimeService.global_service is None
+                else QiskitRuntimeService.global_service
+            )
+            self._backend = self._service.backend(mode)
+        elif get_cm_session():
+            self._mode = get_cm_session()
+            self._service = self._mode.service
+            self._backend = self._service.backend(  # type: ignore
+                name=self._mode.backend(), instance=self._mode._instance
+            )
+        else:
+            raise ValueError("A backend or session must be specified.")
+
+        self._set_options(options)
+
+    @property
+    def options(self) -> NoiseLearnerOptions:
+        """The options in this noise learner."""
+        return self._options
+
+    def run(self, tasks: Iterable[Union[QuantumCircuit, EstimatorPubLike]]) -> RuntimeJobV2:
+        """Submit a request to the noise learner program.
+
+        Args:
+            tasks: An iterable of circuits to run the noise learner program for. Alternatively,
+                pub-like (primitive unified bloc) objects can be specified, such as
+                tuples ``(circuit, observables)`` or ``(circuit, observables, parameter_values)``.
+                In this case, the pub-like objects are converted to a list of circuits, and all
+                the other fields (such as ``observables`` and ``parameter_values``) are ignored.
+
+        Returns:
+            Submitted job.
+
+        """
+        if not all(isinstance(t, QuantumCircuit) for t in tasks):
+            coerced_pubs = [EstimatorPub.coerce(pub) for pub in tasks]
```

Comment:
Is this just for validation?

Reply:
Yes, because they may pass actual pubs or tuples, and I want to avoid logic to distinguish between all these cases